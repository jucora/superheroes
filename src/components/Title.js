import React from "react";
import { NavLink, HashRouter as Router } from "react-router-dom";

const Title = () => (
  <Router>
    <div className="title row">
      <header>
        <h1 data-testid="mainTitle">SUPERHEROES</h1>
        <div data-testid="title-subtitle" className="title-subtitle">
          Search and get information about your favorite superhero!
        </div>
        <div>
          <NavLink
            data-testid="listLink"
            className="nav-link"
            activeClassName="nav-link-active"
            exact
            to="/"
          >
            List
          </NavLink>
          <NavLink
            data-testid="aboutLink"
            className="nav-link"
            activeClassName="nav-link-active"
            to="/about"
          >
            About
          </NavLink>
        </div>
      </header>
    </div>
  </Router>
);

export default Title;
