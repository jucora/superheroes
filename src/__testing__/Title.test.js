import Title from '../components/Title';

it('should take a snapshot', () => {
  const wrapper = shallow(<Title />);
  expect(wrapper).toMatchSnapshot();
});

it('should display a main title', () => {
  const wrapper = mount(<Title />);
  const text = wrapper.find('h1').text();
  expect(text).toEqual('SUPERHEROES');
});

it('should display a subtitle', () => {
  const wrapper = mount(<Title />);
  const text = wrapper.find('.title-subtitle').text();
  expect(text).toEqual(
    'Search and get information about your favorite superhero!',
  );
});

it('should have a link to redirect to hero List', () => {
  const wrapper = mount(<Title />);
  const text = wrapper.find('NavLink').at(0).text();
  expect(text).toEqual('List');
});

it('should have a link to redirect to the Author section', () => {
  const wrapper = mount(<Title />);
  const text = wrapper.find('NavLink').at(1).text();
  expect(text).toEqual('About');
});
