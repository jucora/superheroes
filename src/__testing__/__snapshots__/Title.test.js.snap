// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <header>
      <h1
        data-testid="mainTitle"
      >
        SUPERHEROES
      </h1>
      <div
        class="title-subtitle"
        data-testid="title-subtitle"
      >
        Search and get information about your favorite superhero!
      </div>
      <div>
        <a
          aria-current="page"
          class="nav-link nav-link-active"
          data-testid="listLink"
          href="#/"
        >
          List
        </a>
        <a
          class="nav-link"
          data-testid="aboutLink"
          href="#/about"
        >
          About
        </a>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
